tachyon/base/strings/string_util.h

```diff
@@ -164,6 +164,35 @@ std::string Container2DToString(const Container& data) {
   return ss.str();
 }
 
+template <typename Container>
+std::string ContainerToHexString(const Container& data, bool pad_zero = false) {
+  std::stringstream ss;
+  ss << "[";
+  for (size_t i = 0; i < std::size(data); ++i) {
+    ss << data[i].ToHexString(pad_zero);
+    if (i != std::size(data) - 1) {
+      ss << ", ";
+    }
+  }
+  ss << "]";
+  return ss.str();
```

Comment:
How about this?

```suggestion
  size_t end = std::size(data) - 1;
  for (size_t i = 0; i < end; ++i) {
    ss << data[i].ToHexString(pad_zero) << ", ";
  }
  ss << data[end].ToHexString(pad_zero) << "]";
  return ss.str();
```

Reply:
This would result in looping through the entire range of `size_t` if `data` is empty

Reply:
Then how about this? I wanted to remove the if-statement within the for-statement.

```cpp
std::string ContainerToHexString(const Container& data, bool pad_zero = false) {
  size_t end = std::size(data);

  if (end == 0) return "[]";

  --end; // If it doesn't look good, fix it accordingly.
  std::stringstream ss;
  ss << "[";

  for (size_t i = 0; i < end ; ++i) {
    ss << data[i].ToHexString(pad_zero) << ", ";
  }

  ss << data[end].ToHexString(pad_zero) << "]";
  return ss.str();
}
```